packages/medusa/src/services/shipping-option.ts

```diff
@@ -450,35 +553,45 @@ class ShippingOptionService extends BaseService {
    * will throw errors if metadata or product updates are attempted.
    * @param {string} optionId - the id of the option. Must be a string that
    *   can be casted to an ObjectId
-   * @param {object} update - an object with the update values.
-   * @return {Promise} resolves to the update result.
+   * @param {ShippingOptionUpdate} update - an object with the update values.
+   * @return {Promise<ShippingOption>} resolves to the update result.
    */
-  async update(optionId, update) {
+  async update(
+    optionId: string,
+    update: ShippingOptionUpdate
+  ): Promise<ShippingOption> {
     return this.atomicPhase_(async (manager) => {
       const option = await this.retrieve(optionId, {
         relations: ["requirements"],
       })
 
-      if ("metadata" in update) {
+      if (typeof update.metadata !== "undefined") {
         option.metadata = await this.setMetadata_(option, update.metadata)
       }
 
-      if (update.region_id || update.provider_id || update.data) {
+      if (
+        typeof update.region_id !== "undefined" ||
+        typeof update.provider_id !== "undefined" ||
+        typeof update.data !== "undefined"
+      ) {
         throw new MedusaError(
           MedusaError.Types.NOT_ALLOWED,
           "Region and Provider cannot be updated after creation"
         )
       }
 
-      if ("is_return" in update) {
+      if (
+        typeof update.is_return !== "undefined" &&
+        update.is_return !== undefined
+      ) {
         throw new MedusaError(
           MedusaError.Types.NOT_ALLOWED,
           "is_return cannot be changed after creation"
         )
       }
 
-      if ("requirements" in update) {
-        const acc = []
+      if (typeof update.requirements !== "undefined" && update.requirements) {
```

Comment:
If the requirements are there we can go ahead and update
```suggestion
      if (typeof update.requirements !== "undefined") {
```